pyface/ui/qt4/dialog.py

```diff
@@ -138,6 +148,18 @@ def _show_modal(self):
         retval = dialog.exec_()
         return _RESULT_MAP[retval]
 
+    # -------------------------------------------------------------------------
+    # 'IWidget' interface.
+    # -------------------------------------------------------------------------
+
+    def destroy(self):
+        if self.control is not None:
```

Comment:
If the list is nonempty, it would be best to disconnect them than keeping the references.
(Likewise for the other method)